default/AI/ResearchAI.py

```diff
@@ -69,434 +364,53 @@ def generate_research_orders():
-                        print msg
-                        if report_adjustments:
-                            chat_human(msg)
-                research_queue_list = get_research_queue_techs()
-    #
-    # check to accelerate quant net
-    if False:  # disabled for now, otherwise just to help with cold-folding / organization
-        if (foAI.foAIstate.aggression > fo.aggression.cautious) and (ColonisationAI.empire_status.get('researchers', 0) >= 40):
-            if not tech_is_complete("LRN_QUANT_NET"):
-                insert_idx = num_techs_accelerated  # TODO determine min target slot if reenabling
-                for qnTech in ["LRN_NDIM_SUBSPACE", "LRN_QUANT_NET"]:
-                    if qnTech not in research_queue_list[:insert_idx + 2] and not tech_is_complete(qnTech):
-                        res = fo.issueEnqueueTechOrder(qnTech, insert_idx)
-                        num_techs_accelerated += 1
-                        insert_idx += 1
-                        msg = "Empire has many researchers, so attempted to fast-track %s (got result %d) on turn %d" % (qnTech, res, fo.currentTurn())
-                        print msg
-                        if report_adjustments:
-                            chat_human(msg)
-                research_queue_list = get_research_queue_techs()
+    possible = sorted(priorities.keys(), key=priorities.__getitem__)
 
-    #
-    # if we own a blackhole, accelerate sing_gen and conc camp
-    if True:  # just to help with cold-folding / organization
-        if (fo.currentTurn() > 50 and len(AIstate.empireStars.get(fo.starType.blackHole, [])) != 0 and
-                foAI.foAIstate.aggression > fo.aggression.cautious and not tech_is_complete(AIDependencies.PRO_SINGULAR_GEN) and
-                tech_is_complete(AIDependencies.PRO_SOL_ORB_GEN)):
-            # sing_tech_list = [ "LRN_GRAVITONICS" , "PRO_SINGULAR_GEN"]  # formerly also "CON_ARCH_PSYCH", "CON_CONC_CAMP",
-            sing_gen_tech = fo.getTech(AIDependencies.PRO_SINGULAR_GEN)
-            sing_tech_list = [pre_req for pre_req in sing_gen_tech.recursivePrerequisites(empire_id) if not tech_is_complete(pre_req)]
-            sing_tech_list += [AIDependencies.PRO_SINGULAR_GEN]
-            for singTech in sing_tech_list:
-                if singTech not in research_queue_list[:num_techs_accelerated+1]:
-                    res = fo.issueEnqueueTechOrder(singTech, num_techs_accelerated)
-                    num_techs_accelerated += 1
-                    msg = "have a black hole star outpost/colony, so attempted to fast-track %s, got result %d" % (singTech, res)
-                    print msg
-                    if report_adjustments:
-                        chat_human(msg)
-            research_queue_list = get_research_queue_techs()
+    print "Research priorities"
+    print "    %25s %8s %8s %s" % ("Name", "Priority", "Cost", "Missing Prerequisties")
+    for tech_name in possible[len(possible)-10:]:
```

Comment:
You show last 10 techs?
You can `possible[-10:]` because if `len(possilbe) < 10` you will got not what you want.